memgraph_versioned_docs/version-2.2.1/how-to-guides/streams/kafka/implement-transformation-module.md

```diff
@@ -201,44 +200,94 @@ const std::string query_part_2{"', payload: '"};
 const std::string query_part_3{"', topic: '"};
 const std::string query_part_4{"'})"};
 
-std::string create_query(const mgp_message &message) {
-  return query_part_1 + std::to_string(mgp_message_timestamp(&message)) +
-         query_part_2 +
-         std::string{mgp_message_payload(&message),
-                     mgp_message_payload_size(&message)} +
-         query_part_3 + mgp_message_topic_name(&message) + query_part_4;
+std::string create_query(mgp_message &message, struct mgp_result *result) {
+  int64_t timestamp{0};
+  if (mgp_error::MGP_ERROR_NO_ERROR !=
+      mgp_message_timestamp(&message, &timestamp)) {
+    throw "Internal error!";
+  }
+
+  const char *payload;
+  if (mgp_error::MGP_ERROR_NO_ERROR !=
+      mgp_message_payload(&message, &payload)) {
+    throw "Internal error!";
+  }
+
+  size_t payload_size;
+  if (mgp_error::MGP_ERROR_NO_ERROR !=
+      mgp_message_payload_size(&message, &payload_size)) {
+    throw "Internal error!";
+  }
+
+  const char *topic_name;
+  if (mgp_error::MGP_ERROR_NO_ERROR !=
+      mgp_message_topic_name(&message, &topic_name)) {
+    throw "Internal error!";
+  }
+
+  return query_part_1 + std::to_string(timestamp) + query_part_2 +
+         std::string{payload, payload_size} + query_part_3 + topic_name +
+         query_part_4;
 }
 
-void my_c_transformation(const struct mgp_messages *messages,
-                         const struct mgp_graph *, struct mgp_result *result,
-                         struct mgp_memory *memory) {
+void my_c_transformation(struct mgp_messages *messages, mgp_graph *,
+                         mgp_result *result, mgp_memory *memory) {
+
+  mgp_value *null_value = NULL;
+  if (mgp_error::MGP_ERROR_NO_ERROR !=
+      mgp_value_make_null(memory, &null_value)) {
+    return;
+  }
 
-  auto *null_value = mgp_value_make_null(memory);
   try {
-    auto messages_size = mgp_messages_size(messages);
+    size_t messages_size;
+    if (mgp_error::MGP_ERROR_NO_ERROR !=
+        mgp_messages_size(messages, &messages_size)) {
+      return;
+    }
```

Comment:
If you return here, then `null_value` will be leaked. Maybe get the size of `messages` before creating `null_value`.

Reply:
Good catch, thanks